tests/parser/tests_IStreamQuadIterator.cpp

```diff
@@ -713,4 +713,14 @@ TEST_SUITE("IStreamQuadIterator") {
         }
 
     }
+
+    TEST_CASE("EOF in the middle") {
+        // validate fix of https://gitlab.com/drobilla/serd/-/issues/32
+        // previously this triggered an assertion
+
+        std::istringstream iss{"<htt"};
+        for (IStreamQuadIterator qit{iss}; qit != std::default_sentinel; ++qit) {
+            CHECK_EQ(qit->error().message, "unexpected end of file");
+        }
```

Comment:
Could you also please add test cases for bnodes and literals?

Reply:
Also changed to `CHECK_FALSE(qit->has_value())` as the specific error message really doesn't matter to us